### ISSUE TITLE
UADE playback issues

### ISSUE THREAD

**Issue by bryc** [arnaud-neny/rePlayer#16]
Here is a collection of "Amiga" songs intended for DeliTracker / DeliPlayer / UADE (available for many years at chiptune.com): [amiga.zip](https://github.com/arnaud-neny/rePlayer/files/13637382/amiga.zip)

I noticed a few issues with how rePlayer handles these files (Tested on Win 10).

Most of these files tend to work on this web player: https://hippoplayer.se/WB.html

**Some files freeze the program when attempting to play:**

prince.hp (could be corrupt, not sure)

**Some files don't play, or don't play correctly:**

For example, a_prehistoric_tale.soc used to work fine in DeliPlayer years ago, but not UADE. 

Other files that have similar issues:

astaroth.sog
spacedemo.soc
stormlord.soc

I think it requires a specific replayer binary that is not packaged with UADE.

It might be useful to expose the UADE binary configurations and perhaps look into other WantedTeam Eagleplayers (http://wt.exotica.org.uk/) and DeliTracker / DeliPlayer replayer binaries here: https://archive.org/download/deliplayer-pro/amigaplayers.zip)

All .hpc (Jochen Hippel COSO) files do not work (These work fine in Delix XMPlay plugin).
_Edit: .hpc works if renamed to .hipc_

bubble_bobble.dw is detected but cannot be opened (David Whittaker format), possibly requires specific replayer binary - but I'm not sure. Might be a corrupt or invalid file.

All .amad files (AY Amadeus / PlayAY) files don't work but I think those are AY files? I forget if DeliTracker was supposed to play files like these.

**Some files crash the program:**

The file ackerlight.st seems to crash immediately at random. It's an ZX Spectrum tracker file (maybe added by mistake by pack author) detected by zxtune.

**Comment by arnaud-neny** [comment]
If the player is not accepting files when you drag'n drop, it's because the extension is not registered. You can force it by keeping Ctrl pressed while doing the dra'n drop. But sometimes it's not enough for uade to identify the repay, so you can edit the song to force it: set the replau to uade in the song eeditor, and then you will have a combo with all the amiga replays available. If it's not working, then it's up to the uade team.
It's mostly the same for all other platforms. For the crashes, I'm interested by the files os I can fix it.

**Comment by arnaud-neny** [comment]
Did a lot of fixes based on your playlist, the ackerlight.st is playing now with zxtune, and fix some issues with uade (prince.hp is playing now but it has a bad detection with the number of songs). But it can't still play properly some soc files. I don't have a player handling the .amad, still waiting for zxtune to handle it (there is an old request [there](https://bitbucket.org/zxtune/zxtune/issues/150/ay-amad-support)). For other files not playing with uade, well, it's uade, feel free to report it on their [repository](https://gitlab.com/uade-music-player/uade).

**Comment by bryc** [comment]
Nice work! Glad I could help in a small way. :)

We have been discussing this on the Demozoo#modland discord server in case you are interested.

We have learned that sometimes there are multiple versions of EaglePlayers that cause compatibility issues with some files.

There are 3 versions of Jochen_Hippel_ST, identified by a string in the binary with the year 1997, 2006 and 2008 respectively. I believe you use the 2006 version. The 2008 version seems to have fixed some phase alignment issue and other things.

So it seems that the archive of files I shared was intended for older versions of these players (which are in use with the older DeliPlayer program), and somehow the data format changed in the newer versions UADE uses.

In fact, in some cases there are two versions of the same song, seemingly updated to be compatible with newer versions of the players. For example, [prehistorictale_amiga_STver.zip](https://github.com/arnaud-neny/rePlayer/files/13997965/prehistorictale_amiga_STver.zip) contains versions of "prehistoric_tale.soc" that work in 2006 and 2008 versions of the player, while the other version only works in the 1997 player.

Files like comic_bakery.sog will only work in the newer 2006 and 2008 version of "Jochen_Hippel_ST" eagleplayer, but a_prehistoric_tale.soc, spacedemo.soc and stormlord.soc only work in the 1997 version (aka "Jochen_Hippel_ST_note").

I noticed a similar issue with David Whittaker files (.dw). There are two versions of the file bubble_bobble.dw. The one from that archive doesn't work in UADE, but the version on modland works fine.

Cheers
